### DISCUSSION TITLE
Curiosity: How does Spirit's XML (or I guess TXT) work?

### DISCUSSION THREAD

**Discussion by THEkatinamicrowave** [FunkinCrew/Funkin#3938] [Q&A] [answered]
what is this and what makes it work in the code
![image](https://github.com/user-attachments/assets/9181d510-1c6b-4a34-9d66-a9d2c6ee4a07)

**Comment by moxie-coder** [comment]
offsets from what I understand, it works like every other characters offsets but his is in an txt file for some reason (likely due to him being pixel or some shit I don't know)

**Reply by KoloInDaCrib** [reply, marked as answer]
Those aren't offsets but rather frame data (basically a substitute for the `.xml` file for spirit, made much simpler than an `.xml` file as this doesn't account for frame offsets unlike an `.xml` file). The string before the equals sign is the name of the frame, and the data after the equals sign is:

Number 1: The X-Position of the Frame, in Pixels
Number 2: The Y-Position of the Frame, in Pixels
Number 3: The Width of the Frame, in Pixels
Number 4: The Height of the Frame, in Pixels 

so to analyze, for instance, line 21 `spirit down_0 = 0 0 256 256` it loads the frame with the name `spirit down_0` at `(0, 0)` with the size `256x256` pixels

**Reply by THEkatinamicrowave** [reply]
ok that makes sense
but why not just use the XML
I get that it's simpler, but like

why
was it just a whoopsie on Moawling's end?

**Reply by KoloInDaCrib** [reply]
¯⁠\⁠_⁠(⁠ツ⁠)⁠_⁠/⁠¯
